File: get_icons.js
const fs = require('fs');
const https = require('https');

const download = (url, dest) => {
  return new Promise((resolve, reject) => {
    https.get(url, { headers: { 'User-Agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_7)' } }, (res) => {
      if (res.statusCode === 301 || res.statusCode === 302) {
        return download(res.headers.location, dest).then(resolve).catch(reject);
      }
      const file = fs.createWriteStream(dest);
      res.pipe(file);
      file.on('finish', () => {
        file.close(resolve);
      });
    }).on('error', (err) => {
      fs.unlink(dest, () => reject(err));
    });
  });
};

async function run() {
  const icons = {
    'finder.png': 'https://upload.wikimedia.org/wikipedia/commons/c/c9/Finder_Icon_macOS_Big_Sur.png',
    'vscode.svg': 'https://upload.wikimedia.org/wikipedia/commons/9/9a/Visual_Studio_Code_1.35_icon.svg',
    'terminal.png': 'https://upload.wikimedia.org/wikipedia/commons/b/b3/Terminalicon2.png',
    'settings.png': 'https://upload.wikimedia.org/wikipedia/commons/e/ec/System_Preferences_icon.png',
    'activity_monitor.png': 'https://upload.wikimedia.org/wikipedia/commons/e/eb/Activity_Monitor_Icon.png',
    'docker.svg': 'https://upload.wikimedia.org/wikipedia/commons/4/4e/Docker_%28container_engine%29_logo.svg',
    'services.png': 'https://upload.wikimedia.org/wikipedia/commons/e/ef/Server-icon.png'
  };

  for (const [name, url] of Object.entries(icons)) {
    console.log(`Downloading ${name}...`);
    try {
      await download(url, `apps/web/public/icons/${name}`);
      console.log(`Downloaded ${name}`);
    } catch(e) {
      console.error(e);
    }
  }
}

run();

File: get_icons.py
import requests
import json
import urllib.parse
import os

os.makedirs('apps/web/public/icons', exist_ok=True)

apps = {
    'finder.png': 'macos big sur finder icon png wikimedia',
    'vscode.png': 'macos big sur vscode icon png wikimedia',
    'terminal.png': 'macos big sur terminal icon png wikimedia',
    'settings.png': 'macos big sur system preferences icon png wikimedia',
    'activity_monitor.png': 'macos big sur activity monitor icon png wikimedia',
    'docker.png': 'docker macos big sur icon png wikimedia',
    'automator.png': 'automator macos big sur icon png wikimedia',
    'services.png': 'network utility macos big sur icon png wikimedia'
}

for name, query in apps.items():
    if os.path.exists(f'apps/web/public/icons/{name}') and os.path.getsize(f'apps/web/public/icons/{name}') > 10000:
        continue
        
    print(f"Searching {name}...")
    url = f"http://127.0.0.1:8888/search?q={urllib.parse.quote(query)}&categories=images&format=json"
    try:
        r = requests.get(url, timeout=5)
        data = r.json()
        for res in data.get('results', []):
            img_url = res.get('img_src')
            if not img_url: continue
            if 'wikimedia' not in img_url and 'github' not in img_url and 'mac' not in img_url: continue
            
            print(f"  Trying {img_url}")
            try:
                img_res = requests.get(img_url, headers={'User-Agent': 'Mozilla/5.0'}, timeout=5)
                if img_res.status_code == 200 and 'image' in img_res.headers.get('content-type', ''):
                    with open(f'apps/web/public/icons/{name}', 'wb') as f:
                        f.write(img_res.content)
                    print(f"  Downloaded {name}!")
                    break
            except Exception as e:
                print(f"  Failed: {e}")
    except Exception as e:
        print(f"Search failed for {name}: {e}")

